src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1258,29 +1338,22 @@ def cached_test_function(self, buffer):
     def debug(self, msg):
         self.__engine.debug(msg)
 
-    def shrink(self):
-        """Run the full set of shrinks and update shrink_target.
+    def is_trivial(self):
+        """We assume that if an all-zero block of bytes is an interesting
+        example then we're not going to do better than that.
 
-        This method is "mostly idempotent" - calling it twice is unlikely to
-        have any effect, though it has a non-zero probability of doing so.
+        This might not technically be true: e.g. for integers() |
+        booleans() the simplest example is actually [1, 0]. Missing this
```

Comment:
Instead of the "actually", I'd sketch the distinction between generated values and underlying bytes.  In this case we're dealing with the bytes, so a framing like *"for well-formed strategies shrinking bytes will also shrink the generated values, but this is not guaranteed for all strategies (eg <the above>, or `integers().map(lambda x: x - 10**5)`)"* might be more helpful.